### DISCUSSION TITLE
Shorthand for any type

### DISCUSSION THREAD

**Discussion by jayvdb** [juhaku/utoipa#567] [Ideas]
How do you define that a member can contain any type.  i.e `value_type = Value` that results in `type: [string,number,integer,object,array,boolean,null]` ?

This is needed if the endpoint or field can accept any valid JSON.

The docs for `value_type` at https://docs.rs/utoipa/latest/utoipa/derive.ToSchema.html doesnt explain this.

It feels like `Value` might be a good shorthand for this, due to `serde_json::Value` being the Rust mechanism for holding any JSON value.

**Comment by jayvdb** [comment]
I see `self.is("Object")` is the only use of `self.is`, so it looks like there isnt any other "virtual" type.

Putting `serde_json::Value` in the schema list failed for the standard reason:

> the trait `ToSchema<'_>` is not implemented for `Value`.

["web::Json"](https://docs.rs/actix-web/4.3.1/actix_web/web/struct.Json.html) is the actix-web equivalent of raw JSON, and it is valid as input and output of endpoints, and I am sure the other web frameworks have an equivalent of this base layer type.

**Reply by juhaku** [reply]
Yes, `Object` is the only virtual type currently and it can be used in cases the underlying type is not known or is wished to be kept hidden.

**Comment by jayvdb** [comment]
A quick attempt at building this failed because `OneOfBuilder` doesnt support primitive types.  But I was able to build this by adding `SchemaType::Value`. Need to clean it up a bit.

**Reply by jayvdb** [reply]
https://github.com/juhaku/utoipa/pull/568

**Reply by juhaku** [reply]
We can have support for any type, I'll reply to the PR.

**Reply by jayvdb** [reply]
Merged. Once this is released I can finish https://github.com/idubrov/json-patch/pull/27 so that it uses `Value` instead of `Object`, thus more closely matching the JSON-Patch spec.

**Reply by juhaku** [reply]
Cool, I'll make the release. :slightly_smiling_face:
